middleware/vendor/github.com/digitalbitbox/bitbox-wallet-app/LICENSE

```diff
@@ -0,0 +1,211 @@
+Copyright 2018 Shift Devices AG, Switzerland. All rights reserved.
```

Comment:
(Optional) I'd be fine adding the entire `github.com/digitalbitbox/bitbox-wallet-app` as a `vendor`'ed dependency if that's the easiest way to proceed, but I am noticing that we're adding ~231k LoC to the repo with this PR.

Would it make sense to consider either:

a) having a more granular dependency (if available)
b) managing the `github.com/digitalbitbox/bitbox-wallet-app` dependency another way, such as making it a [git subtree](https://github.com/hkjn/src#subtree)?

cc @benma, maybe you have some advice?

Reply:
Currently I pulled the logging package in, since i make extensive use of it, but it does not make much sense to have it in the tree. So either, we pull in everything through go's dependency management (and refer to `github.com/digitalbitbox/bitbox-wallet-app/util/logging`), or we move whatever we need from the `bitbox-wallet-app` repo into our tree and make sure there are no dangling dependencies (as is the case now and the reasons behind why dep ensure pulled in the entire bitbox-wallet-app repo). Would be good to get @KasparEtter 's feedback here as well.

Reply:
It seems you only use logging and errp.go from the app. 

errp is actually just re-exposing the `github.com/pkg/errors` package:

```Go
package errp

import (
	"github.com/pkg/errors"
)

var (
	// New wraps errors.New
	New = errors.New
	// Newf wraps errors.Newf
	Newf = errors.Errorf
	// WithStack wraps errors.WithStack
	WithStack = errors.WithStack
	// Cause wraps errors.Cause
	Cause = errors.Cause
	// Wrap wraps errors.Wrap
	Wrap = errors.Wrap
	// WithMessage wraps errors.WithMessage
	WithMessage = errors.WithMessage
)
``` 

So I suggest just copying this file or using the errors package directly.

As for logging: imho also not worth pulling in the whole app just for this. I suggest to copy the relevant part, and we can think about moving it to a separate repo. Note that this is also only a thin wrapper around the logrus library, which you can also use directly (the app has configurable logging behavior, which is not even needed in the app, much less in BitBox Base, where you can just define where the output should go to).

Reply:
> It seems you only use logging and errp.go from the app.

> So I suggest just copying this file (without the DetailedError/WithContext stuff) or using the errors package directly.

> As for logging: imho also not worth pulling in the whole app just for this. I suggest to copy the relevant part

I agree with all of this; as they say in the Go community, "a little copying is better than a little dependency". Let's just move over the specific code we need for now, and if it seems like it's likely to grow we can consider making it a separate library that both Base and App can depend on.